packages/server/lib/cloud/studio/StudioLifecycleManager.ts

```diff
@@ -183,7 +237,50 @@ export class StudioLifecycleManager {
       listener(studioManager)
     })
 
-    this.listeners = []
+    if (!process.env.CYPRESS_LOCAL_STUDIO_PATH) {
+      this.listeners = []
+    }
+  }
+
+  private setupWatcher ({
+    projectId,
+    cloudDataSource,
+    cfg,
+    debugData,
+  }: {
+    projectId?: string
+    cloudDataSource: CloudDataSource
+    cfg: Cfg
+    debugData: any
+  }) {
+    // Don't setup a watcher if the studio bundle is NOT local
+    if (!process.env.CYPRESS_LOCAL_STUDIO_PATH) {
+      return
+    }
+
+    // Close the watcher if a previous watcher exists
+    if (StudioLifecycleManager.watcher) {
+      StudioLifecycleManager.watcher.removeAllListeners()
+      StudioLifecycleManager.watcher.close().catch(() => {})
+    }
+
+    // Watch for changes to the studio bundle
+    StudioLifecycleManager.watcher = chokidar.watch(path.join(process.env.CYPRESS_LOCAL_STUDIO_PATH, 'server', 'index.js'), {
+      awaitWriteFinish: true,
+    }).on('change', async () => {
+      await this.studioManager?.destroy()
```

Comment:
Let's add a `console.log` here so we have an indicator that the server has reloaded.